lib/std/posix.zig

```diff
@@ -103,7 +106,35 @@ pub const STDIN_FILENO = system.STDIN_FILENO;
 pub const STDOUT_FILENO = system.STDOUT_FILENO;
 pub const SYS = system.SYS;
 pub const Sigaction = system.Sigaction;
-pub const Stat = system.Stat;
+pub const Stat = if (no_stat) struct {
+    // The system does not have a native stat structure. Make one up.
```

Comment:
It is just literally made up. I just made sure it has the fields that a `Stat` structure exposed from this module is expected to have according to `man 2 stat`.

Keep in mind that we're presenting an illusion here. The underlying system literally just doesn't use `struct stat`, but we have to present the illusion that it does because this is a POSIX layer. glibc and musl do the same.

Reply:
Hmm, I see. If it's never used, why not make it void? i.e `{}`.

Reply:
It is used; you're probably confusing the layers here. :slightly_smiling_face: It's `std.os.linux.Stat` that would never be used (if it was defined as anything but a compile error). But `std.posix` exists at a higher level where it needs to present the illusion of the kernel speaking full POSIX even if it technically doesn't. So `std.posix.Stat` must exist.

Reply:
Ohhhhh, I understand now. Apologies, this makes sense now.